src/Validation/ValidatorAwareTrait.php

```diff
@@ -95,8 +95,16 @@ public function validator($name = null, Validator $validator = null)
         }
 
         if ($validator === null) {
-            $validator = new Validator();
-            $validator = $this->{'validation' . ucfirst($name)}($validator);
+            $className = isset($this->_validatorClass) ? $this->_validatorClass : 'Validator';
+            if ($className === 'Validator') {
+                $className = '\Cake\Validation\Validator';
+            }
+            $className = App::className($className, 'Validation/Validator', 'Validator');
+            if (!$className) {
+                throw new Exception\MissingValidatorException(['class' => $className]);
+            }
+
+            $validator = $this->{'validation' . ucfirst($name)}(new $className);
```

Comment:
> In the past I've handled this by returning a new validator from the validation hook method. 

That would still cause an instance of `Cake\Validator\Validator` unnecessarily created each time. Plus `Table::validation<name>()` methods are called before the `Model.buildValidator` is dispatched, so the rules added to the validator in  `MyTable::validationDefault()` would have to be fetched and added again to the new validator instance one intends to return from the event callback.

> Are you wanting to have a custom validator class for all the validators in a table?

Yes, I going to intend to make and use a custom validator which implements the same feature as proposed in PR #5780.